data/scripts/creaturescripts/modal_window_helper.lua

```diff
@@ -0,0 +1,222 @@
+local ModalWindowAutoID = 10000
+
+if not __ModalWindow then
+    __ModalWindow = ModalWindow
+    ModalWindows = {}
+end
+
+ModalWindow = {}
+ModalWindow.__index = ModalWindow
+
+function ModalWindow.new(self, ...)
+    if type(self) ~= "table" then
+        return __ModalWindow(self, ...)
+    end
+
+    self.modalWindowId = 0
+    self.buttons = {}
+    self.choices = {}
+    self.using = 0
+    return setmetatable(self, ModalWindow)
+end
+
+setmetatable(ModalWindow, {
+    __call = function (self, ...)
+        return ModalWindow.new(...)
+    end
+})
+
+function ModalWindow:setTitle(title)
+    self.title = title
+    return true
+end
+
+function ModalWindow:setMessage(message)
+    self.message = message
+    return true
+end
+
+function ModalWindow:addButton(name, callback)
+    if self.buttons[name] then
+        io.write("ModalWindow: Button with name '" .. name .. "' already exists.")
+        return false
+    end
+
+    local id = #self.buttons + 1
+    local button = { id = id, name = name, callback = callback }
+    self.buttons[id] = button
```

Comment:
Wouldnt this cause problems when doing an `addButton('test')` and then `addButton(0)`?